mismi-s3/src/Mismi/S3/Internal.hs

```
@@ -94,5 +95,5 @@ withFileSafe f1 run = do
   uuid <- liftIO nextRandom >>= return . toString
   let f2 = takeDirectory f1 <> "/" <> "." <> takeFileName f1 <> "." <> uuid
   onException
-    (run f2 >>= \a -> liftIO $ whenM (doesFileExist f2) (renameFile f2 f1) >> pure a)
+    (run f2 >>= \a -> liftIO (handle (\e -> if isDoesNotExistError e then return () else throwM e) $ renameFile f2 f1) >> return a)
```

Either needs some error type to be returned or just let it propagate